Library/Homebrew/cmd/update.rb

```diff
@@ -227,6 +218,15 @@ def pull!(options = {})
       @initial_branch = ""
     end
 
+    unless system "git", "diff", "--quiet"
```

Comment:
Now that this is touched, I think a safer way to check for a pristine working copy is to use `git status --untracked-files=all --porcelain` (as is also done in `brew doctor`, see [here](https://github.com/Homebrew/homebrew/blob/ad6f9d1bb856499f8166468c9d96192c44b00f9b/Library/Homebrew/cmd/doctor.rb#L1073)). The current approach will fail to detect staged, but not yet committed changes, and untracked files.